az-core/src/main/java/azkaban/Constants.java

```diff
@@ -220,6 +220,7 @@
     public static final String EXECUTORINFO_REFRESH_MAX_THREADS =
         "azkaban.executorinfo.refresh.maxThreads";
     public static final String MAX_DISPATCHING_ERRORS_PERMITTED = "azkaban.maxDispatchingErrors";
+    public static final String SLEEP_AFTER_DISPATCH_FAILURE = "azkaban.sleepAfterDispatchFailure";
```

Comment:
i feel making this parameter as configurable exposes complexity to azkaban users. Can you think any situation where the parameter needs to be changed?

Reply:
For quicker unit tests. Users shouldn’t have any reason to customize it AFAIK. But it’s not totally absurd to think that in some setups it could even be useful to be able to change this.

Reply:
do you have some examples of such setups?

Reply:
I guess there are two branches:

1. Delay minimization
   - This is the one that makes more sense probably
   - User is scheduling a lot of executions all the time and wants to minimize the dispatching delay due to any temporary dispatch failures (caused by executor replacement, for example).
   - For this to be used effectively in executor replacement, assuming that azkaban-web has been modified to automatically refresh executors without having to restart it (discussion is going on about that with @jamiesjc).
2. Call throttling
   - User wants to use a longer sleep to avoid a high call rate on executors if dispatching keeps failing.
   - This seems more artificial because the dispatch call alone won't probably be too expensive – it would make more sense to increase the max number of dispatch attempts than to increase the sleep time between attempts.

But as I wrote _it’s not totally absurd_, I mean that most likely users won't ever need to touch these. But I would like to have them as properties like this so that they can be customized for unit tests. Other option would be to add setters with package visibility, but that's not as clean IMHO.

Reply:
Nope, it makes the tests slower if there's some sleeping involved. Tests should never sleep.

Reply:
thanks, i don't have very strong opinion, so up to your preference.
One thing I want to add is we should add time unit in property name to avoid confusion.